### PR TITLE
IBX-3110: Added `version` parameter validation in `DownloadController`

### PR DESCRIPTION
**Pull request by barw4**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-3110](https://issues.ibexa.co/browse/IBX-3110)
| **Type**                                   | bug
| **Target Ibexa version** | `v3.3`
| **BC breaks**                          | no

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [ ] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/engineering-team`).

### CODE REVIEW

File: eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php [hunk 1]
```diff
@@ -47,7 +47,11 @@ public function __construct(ContentService $contentService, IOServiceInterface $
     public function downloadBinaryFileAction($contentId, $fieldIdentifier, $filename, Request $request)
     {
         if ($request->query->has('version')) {
-            $content = $this->contentService->loadContent($contentId, null, $request->query->get('version'));
+            $version = $request->query->get('version');
+            if (!is_int($version)) {
+                throw new NotFoundException('File', $filename);
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php:55]: While at it, could you fix method PHPDoc so it reports that throw (it's incorrect 2x return right now)

**Review by alongosz — CHANGES_REQUESTED** [review]

File: eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php [hunk 2]
```diff
@@ -42,12 +42,19 @@ public function __construct(ContentService $contentService, IOServiceInterface $
      * @param \Symfony\Component\HttpFoundation\Request $request
      *
      * @return \eZ\Bundle\EzPublishIOBundle\BinaryStreamResponse
-     * @return \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentValue
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php]: we can't report non-API exception here, it needs to be in API namespace (or SPI if available).

**Review by alongosz — CHANGES_REQUESTED** [review]

**Review by alongosz — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

**Review by glye — APPROVED** [review]
Does it work as expected if version is zero or negative? If yes, I approve :)

**Comment by barw4** [comment]
Actually, I had to cast it to `int` as a request gives `string`.

**Review by alongosz — APPROVED** [review]

**Review by konradoboza — APPROVED** [review]

File: eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php [hunk 3]
```diff
@@ -42,12 +42,19 @@ public function __construct(ContentService $contentService, IOServiceInterface $
...
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
      */
     public function downloadBinaryFileAction($contentId, $fieldIdentifier, $filename, Request $request)
     {
         if ($request->query->has('version')) {
-            $content = $this->contentService->loadContent($contentId, null, $request->query->get('version'));
+            $version = (int) $request->query->get('version');
+            if ($version <= 0) {
```
Comment (glye) [inline eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php:54]: Won't the version now be 0 if not specified in the request? Then all those requests will fail.

Comment (barw4) [reply eZ/Publish/Core/MVC/Symfony/Controller/Content/DownloadController.php:54]: No, because you have `if ($request->query->has('version')) {` there

**Review by glye — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Pull request merged**